samcli/commands/build/build_context.py

```diff
@@ -265,3 +265,27 @@ def _collect_single_buildable_layer(
             raise MissingBuildMethodException(f"Build method missing in layer {resource_identifier}.")
 
         resource_collector.add_layer(layer)
+
+    @staticmethod
+    def _is_function_buildable(function: Function):
+        # no need to build inline functions
+        if function.inlinecode:
+            LOG.debug("Skip building inline function: %s", function.full_path)
+            return False
+        # no need to build functions that are already packaged as a zip file
+        if isinstance(function.codeuri, str) and function.codeuri.endswith(".zip"):
```

Comment:
should we add a strip() before checking the endswith?

Reply:
the `codeuri` is from yaml, so we don't need to.